### PR TITLE
Pipelines

### PR DESCRIPTION
**Pull request by dennisnewel**
Added a new section about working with pipelines. This is meant as a place where you can go to either customize a generated pipeline (via import or quick start) or create a new pipeline from scratch.

This content is by no means all encompassing, but will hopefully work as a starting point for more

As there's a separate page just on environment variables, i moved the table of default environment variables to that page and updated internal references to it

### CODE REVIEW

**Comment by dennisnewel** [comment]
@abayer @jha-cloudbees this is the PR i mentioned earlier today. Would be great if you could have a look at it and let me know if there are changes needed

File: content/en/docs/using-jx/pipelines/newpipeline.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+buildPack: none
+pipelineConfig:
+ pipelines:
+   pullRequest:
+   release:
+```
+
+For the `pullRequest` pipeline you define the stages and steps that Jenkins X will execute for each pull request, and `Release` is the same, just for what should happen to create the final product (whether a k8s app, docker image, static website, etc.)
+
+You can also specify a `feature` pipeline, for processing merges to a feature branch. Though note that the [accelerate book](/docs/overview/accelerate/) recommends against long term feature branches. Instead consider using trunk based development which is a practice of high performing teams.
```
Comment (jha-cloudbees) [inline content/en/docs/using-jx/pipelines/newpipeline.md]:
```suggestion
You can also specify a `feature` pipeline, for processing merges to a feature branch. However, note that the [Accelerate book](/docs/overview/accelerate/) recommends against long term feature branches. Instead, consider using trunk based development which is a practice of high performing teams.
```

**Comment by dennisnewel** [comment]
Thanks @jha-cloudbees ! i'll merge it in once the build runs cleanly

File: content/en/docs/using-jx/pipelines/envvars.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    release:
+      pipeline:
+        stages:
+          steps:
+            environment:
+              - name: STEP_VAR
+                value: A value for the step level variable
+```
+
+Each level will add their variables to the variables defined in the level(s) above them, or potentially replace a variable if the same name is used
```
[the remark this answers is not in the document]
Comment (dennisnewel) [reply content/en/docs/using-jx/pipelines/envvars.md]: strange how GH says this is outdated while not actually having been applied. Added the period in a new commit

**Pull request merged**